apis/cpi/v1alpha1/cpiconfig_types.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CPIConfigSpec defines the desired state of CPIConfig
+type CPIConfigSpec struct {
+	VSphereCPI VSphereCPI `json:"vsphereCPI"`
+}
+
+type NSXTRoute struct {
+	// NSX-T T0/T1 logical router path
+	// +kubebuilder:validation:Optional
+	RouterPath string `json:"routerPath"`
+}
+
+type NSXT struct {
+	// A flag that enables pod routing
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	PodRoutingEnabled bool `json:"podRoutingEnabled"`
+
+	// Route configuration for NSXT
+	// +kubebuilder:validation:Optional
+	Routes *NSXTRoute `json:"routes"`
+
+	// A secret reference that contains Nsx-T login credentials to access NSX-T
+	// consists of the field username and password
+	NSXTCredentialsRef v1.SecretReference `json:"nsxTCredentialsRef"`
+
+	// The NSX-T server
+	// +kubebuilder:validation:Optional
+	Host string `json:"host"`
```

Comment:
I am wondering why this is optional. Can NSXT config work without host info?

Reply:
It should be required https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/nsxt/config/config_yaml.go#L80

Reply:
from the discussion on Friday:

for NSXT related configurations, it should be optional at the API level but required for the CPI package. If user doesn't specify NSXT in the CPIConfig, the controller should read it from the cluster topology, defined at https://github.com/vmware-tanzu/tanzu-framework/blob/fc958cc58c52fd8a178135541fce70ccf39276eb/addons/controllers/testdata/test-cluster-bootstrap-1.yaml#L31

@vijaykatam @blc1996  could you plz confirm?

Reply:
Technically CPI doesn't need NSXT, it is only required for generating data value when NSXT config is provided via cluster variables.

Reply:
`NSXT_MANAGER_HOST` should be available in cluster variable

Reply:
yes, CPI is currently deriving it from `NSXT_MANAGER_HOST` cluster variable. This CPIConfig field is merely a way for API user to override the derived field.